src/lib/fragments/Accordion.react.js

```diff
@@ -0,0 +1,81 @@
+        <div className="govuk-accordion" data-module="govuk-accordion" id="accordion-default">
+          <div className='govuk-accordion__controls'>
+            <button type='button' className='govuk-accordion__show-all' onClick={this.showOrHideAllAccordionSections}>
+            <span className= {this.state.allSectionsAreOpen ? "govuk-accordion-nav__chevron" : "govuk-accordion-nav__chevron govuk-accordion-nav__chevron--down"} ></span>
+            <span className="govuk-accordion__section-toggle-text"> {this.state.allSectionsAreOpen ? "Hide all sections" : "Show all sections"} </span>
+            </button>
+          </div>
+          {this.props.children.map((accordionSectionContent, index) => this.renderAccordionSection(index, accordionSectionContent, this.state.sectionsOpen[index]))}
+        </div>
+      </div>
+    )
+  }
+
+  renderAccordionSection(index, accordionSectionContent, sectionIsOpen) {
+    const { expandedClass="govuk-accordion__section--expanded", collapsedClass="govuk-accordion__section", ShowString = "Show", HideString = "Hide", accordionHeadings} = this.props
+      return (
+          <div className={sectionIsOpen ? collapsedClass + ' ' + expandedClass : collapsedClass}>
+            <div className="govuk-accordion__section-header">
+              <h2 className="govuk-accordion__section-heading">
+                <button type="button" aria-controls={`accordion-default-content-${index}`} className="govuk-accordion__section-button" id={`accordion-default-heading-${index}`} aria-expanded={sectionIsOpen} onClick={() => this.openOrCloseAccordionSection(index)}>
+                  <span className="govuk-accordion__section-heading-text" id={`accordion-default-heading-${index}`}>
+                    <span className="govuk-accordion__section-heading-text-focus"> {accordionHeadings[index]} 
+                    </span>
+                  </span> 
+                  <span className="govuk-visually-hidden govuk-accordion__section-heading-divider"></span>
+                  <span className="govuk-accordion__section-toggle" data-nosnippet>
+                    <span className="govuk-accordion__section-toggle-focus">
+                      <span className="moj-side-navigation__item--collapsed">
+                      </span>
+                    </span>
+                    <span className= {sectionIsOpen ?  "govuk-accordion-nav__chevron" : "govuk-accordion-nav__chevron govuk-accordion-nav__chevron--down"} ></span>
+                    <span className="govuk-accordion__section-toggle-text"> {sectionIsOpen ? HideString : ShowString} 
+                      <span className="govuk-visually-hidden"> this section</span>
+                    </span>
+                  </span>
+                </button>
+              </h2>
+            </div>
+            <div id={`accordion-default-heading-${index}`} className="govuk-accordion__section-content" aria-labelledby={`accordion-default-heading-${index}`}>
```

Comment:
This will cause an issue having ID based purely on index. If the component is used twice in a page this will cause validation issue as ID can only be use one.

This is the same of all of the above. You could use a 3rd party node library https://www.npmjs.com/package/uniqid

alternative you can use static property that has an instance count and every time one generate you add to the number and add that as part of the id's.

I hope you dont mind this feedback sorry

Reply:
Thanks, Jordan. So if we install and import uniqid I wonder if we use something like:
{`accordion-default-heading-***`} uniqid(${index}, '-')
Thoughts @duncanc19 @kateriley1 ?

Reply:
Thanks for spotting this Jordan. We have removed the IDs for now as they weren't being used. It's also been flagged in the follow-up work that there are a few methods for doing this properly for the IDs and for the aria-attributes